public/video-ui/src/components/ScheduledLaunch/ScheduledLaunch.js

```diff
@@ -11,89 +17,338 @@ export default class ScheduledLaunch extends React.Component {
   };
 
   state = {
-    selectedDate: null,
+    selectedScheduleDate: null,
+    selectedEmbargoDate: null,
     showDatePicker: false,
-    showScheduleButton: true
+    actionType: null,
+    showScheduleButton: true,
+    showScheduleOptions: false,
+    invalidDateError: null
+  };
+
+  componentWillReceiveProps(nextProps) {
+    const embargo =
+      nextProps.video.contentChangeDetails.embargo &&
+      nextProps.video.contentChangeDetails.embargo.date;
+    const scheduledLaunch =
+      nextProps.video.contentChangeDetails.scheduledLaunch &&
+      nextProps.video.contentChangeDetails.scheduledLaunch.date;
+    this.setState({ selectedScheduleDate: scheduledLaunch || embargo });
+    this.setState({ selectedEmbargoDate: embargo || scheduledLaunch });
+  }
+
+  validateDate = (date, actionType) => {
+    if (!date) {
+      this.setState({ invalidDateError: null });
+      return;
+    }
+
+    const { video: { contentChangeDetails } } = this.props;
+    const scheduledLaunch =
+      contentChangeDetails &&
+      contentChangeDetails.scheduledLaunch &&
+      contentChangeDetails.scheduledLaunch.date;
+    const embargo =
+      contentChangeDetails &&
+      contentChangeDetails.embargo &&
+      contentChangeDetails.embargo.date;
+
+    this.setState({ invalidDateError: null });
+
+    if (!isFutureDate(date)) {
+      this.setState({ invalidDateError: 'Date must be in the future!' });
+    }
+
+    if (
+      actionType === 'schedule' &&
+      embargo &&
+      isFutureDate(date) &&
+      !isSameOrAfter(date, embargo)
+    ) {
+      this.setState({
+        invalidDateError: "Scheduled launch can't be earlier than embargo!"
+      });
+    }
+
+    if (
+      actionType === 'embargo' &&
+      scheduledLaunch &&
+      isFutureDate(date) &&
+      isAfter(date, scheduledLaunch)
+    ) {
+      this.setState({
+        invalidDateError: "Embargo can't be later than scheduled launch!"
+      });
+    }
   };
 
-  saveScheduledLaunch = () => {
+  onSelectOption = actionType => {
+    const date =
+      actionType === 'schedule'
+        ? this.state.selectedScheduleDate
+        : this.state.selectedEmbargoDate;
+    this.validateDate(date, actionType);
+    this.setState({
+      showDatePicker: true,
+      actionType,
+      showScheduleOptions: false
+    });
+  };
+
+  preventPublication = () => {
     const video = this.props.video;
-    this.props.saveVideo(Object.assign({}, video, {
+    const impossiblyDistantDate = 16725225600000;
```

Comment:
I'd put this somewhere more visible, to the top of the file or in the constants folder.

Reply:
in addition to @Reettaphant 's suggestion, I'd either add a comment explaining its value (year 3000) or express it as a moment function `moment('3000-01-01).toString()` (or whatever the correct syntax is...)